src/fprime/common/models/serialize/time_type.py

```diff
@@ -209,20 +205,28 @@ def deserialize(self, data, offset):
         self.__usecs.deserialize(data, offset)
         offset += self.__usecs.getSize()
 
-    def getSize(self):
+    @classmethod
+    def getSize(cls):
         """
         Return the size of the time type object when serialized
 
         Returns:
             The size of the time type object when serialized
         """
         return (
-            self.__timeBase.getSize()
-            + self.__timeContext.getSize()
-            + self.__secs.getSize()
-            + self.__usecs.getSize()
+            U16Type.getSize() + U8Type.getSize() + U32Type.getSize() + U32Type.getSize()
```

Comment:
I would argue to revert this change. For example, if you change `self.__timeBase`'s type to U8, then this won't work.

Reply:
Good catch!

Reply:
This will be fixed when we make the ground system configurable. Until then, this is as it is.